Removed duplicate report

nimrodshn opened the pull request:
Due to #14494 I have removed the the duplicate report and renamed the old one for consistency with widgets.

cc @kbrock @simon3z @moolitayer

nimrodshn commented:
I have tested widgets and report to see changes did not affect them:
Report:
![screenshot from 2017-03-27 12-06-30](https://cloud.githubusercontent.com/assets/8366181/24349414/a9489342-12e7-11e7-835a-6fa94907d44e.png)
Widgets (top two):
![screenshot from 2017-03-27 12-23-53](https://cloud.githubusercontent.com/assets/8366181/24349620/49f97126-12e8-11e7-8a52-8888f78582cc.png)

Everything seems to work fine (to be expected..)

nimrodshn commented:
@miq-bot add_label providers/containers reporting

nimrodshn commented:
and again - thanks to @kbrock for the catch!

moolitayer commented:
@nimrodshn what is the meaning of 
```
miq_group_id: 1
user_id: 1
``` asking since they exists only in the report the remains

Also this exists only in the file you removed, not sure what it does:

```
filename: 170_Configuration Management - Containers/060_Container Groups per Ready Status.yaml
```

moolitayer commented:
@miq-bot add_label bug, providers/containers, reporting
fix #14494

nimrodshn commented:
@moolitayer I think it has something to do with who is allowed to use this report.. to be quite honest I'm not sure as I did not write this report  - Although from previous discussion I had (I think with @zeari) I think it can be removed. Let me check and see :)
As for your'e other question the line you are pointing to is exactly the reason we are deleting that report - it turns out it was a duplicate of the one I renamed - moreover it didn't do anything but draw information from that one do I just deleted it and renamed the old one as to be consistent with other widgets I had brought in.

nimrodshn commented:
@moolitayer also note they exit in other reports (say 020_Nodes by CPU Usage.yaml)

zeari commented:
> @moolitayer I think it has something to do with who is allowed to use this report.. to be quite honest I'm not sure as I did not write this report - Although from previous discussion I had (I think with @zeari) I think it can be removed. Also note they exist in other reports (say 020_Nodes by CPU Usage.yaml).

AFAIK, `miq_group_id` and `user_id` shouldnt be in 'out of the box' reports.

@nimrodshn LGTM (im counting on you that none of the related widgets\reports reports fail)

moolitayer commented:
I'm in favor of going for the other solution @kbrock suggested in  #14494: renaming the duplicate title. Maybe also comment. The reason is that old scheduled reports and widgets would fail. @kbrock @simon3z WDYT?

kbrock commented:
@moolitayer they look like duplicates to me.
Not sure what you would do to change the title for 2 identical reports

nimrodshn commented:
@kbrock I think what @moolitayer is saying is that once I delete one of the reports (either the dup or the original) users who are using it as base for **custom widgets** will lose their widgets since they are based on a report which exist no more.. what do you think?

kbrock commented:
@nimrodshn Do custom widgets to reference reports by name? If so, then there shouldn't be a problem.

If not, then can we write a migration?

nimrodshn commented:
@kbrock Custom widgets (as do "out of the box" widgets) reference reports by name.

kbrock commented:
@nimrodshn Phew. I was concerned that I was totally off here.

>@kbrock I think what @moolitayer is saying is that once I delete one of the reports (either the dup or the original) users who are using it as base for custom widgets will lose their widgets since they are based on a report which exist no more. Although, renaming one of the reports is the same as removing it - So I don't see how this gets around this.. what do you think?

Please do not rename.

The original and custom widgets refer to a report by name.
We don't know which report will be returned. Luckily they have the same data.
After you delete this file, the widgets system refer to a single report with this name.
So nothing will break.
It will actually be more stable.

moolitayer commented:
@kbrock @nimrodshn sorry for holding this back. Just to be sure 
- what about scheduled reports? do we need to migrate that?
- when you say widgets refer to reports by name do you mean `menu_name`, `title`  or filename?

nimrodshn commented:
@moolitayer 
> when you say widgets refer to reports by name do you mean menu_name, title or filename?

I mean `resource_name`.

moolitayer commented:
> I mean resource_name.
> So, if you look at 'Pods per Ready' chart/widget (at product/dashboard/widgets) you will find both are "pointing to" the Pods per Ready report - the Duplicate - that is why I had changed the original name to that - So that I will be able to produce those widgets.

Oh so `resource_name` points to the reports' `title`

nimrodshn commented:
@moolitayer Precisely!
